nautobot/ipam/fields.py

```diff
@@ -1,11 +1,31 @@
 from django.core.exceptions import ValidationError
 from django.db import models
+from django.utils.datastructures import DictWrapper
 from netaddr import AddrFormatError, IPNetwork
 
 from . import lookups, validators
 from .formfields import IPNetworkFormField
 
 
+class VarbinaryIPField(models.BinaryField):
+    """
+    IP network address
+    """
+
+    description = "IP network address"
+
+    def db_type(self, connection):
+        engine = connection.settings_dict["ENGINE"]
+
+        # Use 'bytea' type for Postgres.
+        if "postgres" in engine:
+            return super().db_type(connection)
+
+        # Or 'varbinary' for everyone else.
+        max_length = DictWrapper(self.__dict__, connection.ops.quote_name, "qn_")
+        return "varbinary(%(max_length)s)" % max_length
+
```

Comment:
We can define a `.formfield()` method here that emits a `IPNetworkFormField`. Checkout `BaseIPField.formfield()` below for the pattern you could duplicate!

With that defined the other locations like on `dcim.device.Device.primary_ip4`, there would be no need to explicitly overload that field since it would come directly from the model during form generation.